bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/JPEGDecoder.java

```diff
@@ -2462,22 +2462,18 @@ static void jpeg_calc_output_dimensions (jpeg_decompress_struct cinfo)
 
 	/* Report number of components in selected colorspace. */
 	/* Probably this should be in the color conversion module... */
-	switch (cinfo.out_color_space) {
-		case JCS_GRAYSCALE:
-			cinfo.out_color_components = 1;
-			break;
-		case JCS_RGB:
-		case JCS_YCbCr:
-			cinfo.out_color_components = 3;
-			break;
-		case JCS_CMYK:
-		case JCS_YCCK:
-			cinfo.out_color_components = 4;
-			break;
-		default:			/* else must be same colorspace as in file */
-			cinfo.out_color_components = cinfo.num_components;
-			break;
-	}
+	cinfo.out_color_components = switch (cinfo.out_color_space) {
+	case JCS_GRAYSCALE:
+		yield 1;
```

Comment:
Why not `case JCS_GRAYSCALE -> 1;` ?

Reply:
when I applied your changes I was getting this error "mixing of different case semantics is not allowed". I converted all the cases into one single format. Not sure if I did it right. Can you suggest?

Reply:
the `->` syntax must be used on all fields; it's not allowed to mix `case X: ...` and `case Y -> ...`

Reply:
I have addressed that way.